documentation/book/proc-upgrading-brokers-newer-kafka.adoc

```diff
@@ -0,0 +1,112 @@
+// This module is included in the following assemblies:
+//
+// assembly-upgrading-kafka-versions.adoc
+
+[id='proc-upgrading-brokers-newer-kafka-{context}']
+
+= Upgrading brokers to a newer Kafka version
+
+This procedure describes how to upgrade a {ProductName} Kafka cluster from one version to a higher version; for example 2.0.0 to 2.1.0.
+
+.Prerequisites
+
+* The Cluster Operator, which supports both versions of Kafka, is up and running.
+* A `Kafka` resource to be upgraded.
+* You have checked that your `Kafka.spec.kafka.config` contains no options that are not supported in the version of Kafka that you are upgrading to.
+
+.Procedure
+
+. Consult the table below and determine whether the new Kafka version has a different log message format version than the old version. 
++
+include::snip-kafka-versions.adoc[leveloffset=+1]
++
+If the log message format versions are the same proceed to the next step. 
+Otherwise ensure the `Kafka.spec.kafka.config` has the `log.message.format.version` configured to the default for the old version.
++
+For example, if upgrading from 2.0.0:
++
+[source,yaml]
+----
+apiVersion: v1alpha1
+kind: Kafka
+spec:
+  # ...
+  kafka:
+    version: 2.0.0
+    config:
+      log.message.format.version: "2.0"
+      # ...
+----
++
+NOTE: You must format the value of `log.message.format.version` as a string to prevent it from being interpreted as a number.
++
+If `log.message.format.version` is unset then set it and wait for the resulting rolling restart of the Kafka cluster to complete.
++
+. Change the `Kafka.spec.kafka.version` to specify the new version, but leave the `log.message.format.version` as the old version.
+If the image to be used is different from the image for the given version of Kafka configured in the Cluster Operator's `STRIMZI_KAFKA_IMAGES` then configure the `Kafka.spec.kafka.image` as well.
++
+For example, if upgrading from Kafka 2.0.0 to 2.1.0:
++
+[source,yaml]
+----
+apiVersion: v1alpha1
+kind: Kafka
+spec:
+  # ...
+  kafka:
+    version: 2.1.0 <1>
+    config:
+      log.message.format.version: "2.0" <2>
+      # ...
+----
+<1> This is changed to the new version
+<2> This remains at the old version
+
+. Wait for the Cluster Operator to upgrade the cluster.
+If the old and new versions of Kafka have different interbroker protocol versions, look in the Cluster Operator logs for an `INFO` level message like:
```

Comment:
```suggestion
If both the previous and new versions of Kafka have different interbroker protocol versions, look in the Cluster Operator logs for an `INFO` level message such as:
```

Reply:
It's not an example, it's a pattern! To me, "such as" is quite strongly suggestive of being a literal example. I wonder if "...`INFO` level message matching the following pattern:" (or with the following form) would be clearer, wdyt?

Reply:
The IBM style guide says to not use 'like' to mean 'such as'. 'matching the following pattern' is just more words to say 'such as' :) hehe  I'm not against your wording if you feel strongly about it.